sql/table.cc

```diff
@@ -5227,9 +5227,15 @@ int TABLE::verify_constraints(bool ignore_failure)
       /* yes! NULL is ok, see 4.23.3.4 Table check constraints, part 2, SQL:2016 */
       if ((*chk)->expr->val_int() == 0 && !(*chk)->expr->null_value)
       {
-        my_error(ER_CONSTRAINT_FAILED,
-                 MYF(ignore_failure ? ME_JUST_WARNING : 0), (*chk)->name.str,
-                 s->db.str, s->table_name.str);
+        if ((*(*chk)).get_vcol_type()==VCOL_CHECK_TABLE)
```

Comment:
Uhm, please don't use such syntax, it looks ugly and it doesn't convey the meaning properly.
`(*chk)->get_vcol_type() == VCOL_CHECK_TABLE`

And please watch coding style. However let's rewrite the whole block like in the next comment.